### DISCUSSION TITLE
Plot3D on WSL/jupyter

### DISCUSSION THREAD

**Discussion by galacticStorm** [NanoComp/meep#2424] [Q&A] [answered]
Hi again! Using meep in WSL I have been trying to create a 3D visualization of my meep geomtry using the plot3D function. I came across an example on the [meep website](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#visualizing-3d-structures) (at the bottom), unfortunately it doesn't work. I have installed mayavi (+tested with the ipy and itk backends) and a host of other libraries which the environment was missing, but I can't for the life of me seem to get a visualization in 3D.  I am stuck with the issue: "WARNING: could not determine DPI" as can be seen in the image below.
![image](https://user-images.githubusercontent.com/76227914/222967072-cb6b1a61-4777-4b90-a4a2-e3def062dcc6.png)

I know I recently asked a relatively simple question and I'm very grateful it got answered, I really have spent couple several hours trying to figure this out but I'm just not that experienced, so any help would be super appreciated!

UPDATE: A vispy window opens now too, but its just black.

**Comment by smartalecH** [comment]
(cc @SkandanC who recently overhauled this feature)

**Comment by galacticStorm** [comment]
So I realised its no longer using mayavi, and vispy instead (despite what the docs said). I got vispy installed and use pyglet as its backend + installed [freeglut-dev](https://stackoverflow.com/questions/50446867/importerror-library-glu-not-found)  I also of course have jupyter-rfb too. Currently in Ubuntu 22.04 latest mesa drivers and I can visualize my geometries correctly if I run them as a script in terminal, however with jupyter they still only produce a black canvas. The DPI warning has not gone away (even in the terminal), however in jupyter notebook (only!) I get these additional errors:
![image](https://user-images.githubusercontent.com/76227914/223091312-7b2f9f71-8415-4410-88f6-97ec9644f0ab.png)

**Comment by SkandanC** [comment, marked as answer]
Hello,

This seems like an issue on Vispy's end and not on Meep's end. 
https://github.com/vispy/vispy/issues/2458

**Comment by galacticStorm** [comment]
I see, hope they fix it soon then as executing my files from terminal is a bit annoying. It might also be nice to fix the documentation in the website at the (Python) Tutorial/Basics at "Visualizing 3d Structures"... I'm also going to read the how to contribute doc soon so maybe I'll do it :)

Thanks all for the help!

**Reply by 10UAikawa** [reply]
Hi, I have the same issue, do you know if the problem is fixed and if so, could you explain to me what must be done?

**Reply by 10UAikawa** [reply]
I did everything you did in https://github.com/vispy/vispy/issues/2068 it runs with no errors, but I only get a black screen.
